apps/log/lib/log/model/log.ex

```diff
@@ -61,7 +61,7 @@ defmodule Helix.Log.Model.Log do
     %__MODULE__{}
     |> cast(params, @creation_fields)
     |> validate_required(@required_fields)
-    |> put_primary_key()
+    |> put_change(:log_id, PK.pk_for(__MODULE__))
```

Comment:
Sorry, i don't like javascript, i neve know what `this` could be 😁😁😁😁😁😂😂😂😂😂👏👏👏👏👏👏👌👌👌👌

About that, have you tried simply removing the `log_id` from the changeset (and it's requirement on the Revision model) ?

Because when you assoc something to a changeset (and the schema defines that association), ecto will know what to do on insert (ie: it'll use the returned id and bind them)

Try it :)